### ISSUE TITLE
[CLI Tool] Cadastro de Eventos

### ISSUE THREAD

**Issue by vitorbritto** [frontendbr/eventos#10]

**Comment by vitorbritto** [comment]
Feito! 

cc: @frontendbr/admins

**Comment by eduardojmatos** [comment]
@vitorbritto boa!

**Comment by lucasbsilva** [comment]
+1
@vitorbritto seria interessante visualizar os eventos também.

**Comment by leandrogrillo** [comment]
Eu ia comentar sobre termos um CLI lá na outra issue, mas achei que seria por muitos assuntos em pauta. 

Eu acho a ideia bem interessante, quero ver isso de perto pra poder aprender e acho que poderíamos ter algo que nem o rinaldi fez pro hacker news [link pra quem não viu](https://github.com/rafaelrinaldi/hn-cli)

**Comment by vitorbritto** [comment]
Gosto bastante da ideia de trabalharmos com o [inquirer](https://www.npmjs.com/package/inquirer). Tenho alguns projetos baseado nele. 

Cc: @frontendbr/admins

**Comment by vitorbritto** [comment]
@lucasbsilva com certeza. Eu acredito que podemos começar com um MVP, fazendo o registro e utilizar o [blessed](https://www.npmjs.com/package/blessed) para fazer o output dos eventos no terminal.

**Comment by vitorbritto** [comment]
@leandrogrillo boa referência!

**Comment by leandrogrillo** [comment]
@vitorbritto concordo, temos que ir construindo aos poucos, um MVP seria o mais viável agora e com o tempo ir evoluindo a ferramenta. 

Acho que a demanda de criarmos o evento pelo CLI é o MVP no caso e depois estudamos outras features pra ele.

Até acho que isso pode ser feito sem termos o site pronto.

**Comment by vitorbritto** [comment]
@lendrogrillo acredito que sim. Vamos aguardar as considerações dos colegas para dar início ao projeto.

_Summon:_ @frontendbr/admins

**Comment by leandrogrillo** [comment]
@vitorbritto demoro mano :) vamos ficar no aguardo :D

**Comment by fdaciuk** [comment]
Top a ideia! Acho que a primeira coisa é definir a estrutura do JSON que vai ser gerado.

Como o site vai ser SPA, o ideal era ter um único JSON listando todos os eventos, pra fazer um único request dele. Então, essa CLI vai editar esse JSON e adicionar a nova entrada, correto?

Se tiver mais de um, vamos ter que, de alguma forma, listar todos os JSONs em algum lugar pra saber pra onde fazer o request, já que não rola `filesystem` no browser, e vamos acabar tendo dois trabalhos.

Não sei se vocês tem alguma ideia diferente disso. O que se passa nessas cabecinhas? xD

**Comment by leandrogrillo** [comment]
@fdaciuk na verdade eu pensei em termos um api rest. Acho que nesse caso por nao termos tanta informacao pesada, nao teria problemas em ter mais de uma resquest. Com o webservice daria pra cuidarmos do cache disso e etc, é só uma ideia.

**Comment by fdaciuk** [comment]
@leandrogrillo entendi. É que a ideia inicial do projeto é ele ser um SPA, e ficar hospedado no GitHub pages. O problema de fazer backend disso vai ser a burocra que vai gerar pra manter servidor, vai ter que ter responsáveis pelo server, etc.

Se fizer só como SPA, manda um PR com a alteração no JSON e boa. A parada já tá no ar =)

**Comment by fernandofleury** [comment]
Mas se fizermos com um servidor, permitimos que outros serviços possam se aproveitar dele. Numa ideia bem mais distante, o server pode se tornar um único hub de eventos, e de lá pegamos pelas "categorias" que desejamos.

Honestamente, vejo sendo tão trabalhoso como a CLI, mas com uma escalabilidade muito mais sustentável.

**Comment by fdaciuk** [comment]
Ah sim, isso é certo. Mas eu acho que dá pra pensar nisso depois de ter ao menos uma versão do negócio rodando. Se ficar adicionando muita coisa agora, o negócio nem sai do lugar =)

Minha opinião: acho que a gente pode seguir na ideia do SPA com o CLI, e depois fazer as melhorias: criar uma API pra ser consumida por outros serviços, definir como vai ficar estrutura de server e talz.

Só acho que precisamos ter algo rodando primeiro, senão gera muita ideia e acaba que nada sai do papel.

**Comment by fernandofleury** [comment]
Acho que é um ponto bem válido. Mas temos tantas mãos pra ajudar nesse projeto... será que não sai mesmo?

**Comment by leandrogrillo** [comment]
@fdaciuk foi o que o @vitorbritto falou, vamos partir pro MVP e depois vamos evoluindo, mas já fica a ideia parar partimos em segundo plano.

Só acho que tipo um CLI pra rodar local, só no repo não faria tanto sentido assim. Posso estar falando besteira, mas é o que eu acho.

**Comment by fernandofleury** [comment]
Principalmente com um acompanhamento público do projeto. Acabo achando que fazer uma CLI é jogar um pouco de tempo de produção fora. Um serviço de backend desses fica pronto bem rápido, principalmente se usarem algo como o Loopback.

**Comment by fernandofleury** [comment]
Tudo bem que esse tipo de framework é super opiniado, mas para o que queremos fazer é mais do que o suficiente.

**Comment by vitorbritto** [comment]
@leandrogrillo @fernandofleury vejo num primeiro momento um CLI Tool simplificado, funcional e que consigamos cadastrar os eventos na organização. Concordo sobre a API e em termos um _server-side_ para escalar melhor, mas podemos fazer de forma granular.

> Então, essa CLI vai editar esse JSON e adicionar a nova entrada, correto?

Pensei nisso @fdaciuk. =]

**Comment by eduardojmatos** [comment]
@leandrogrillo @fernandofleury @fdaciuk exatamente a ideia de editar um JSON inicialmente. Depois eu concordo plenamente com o @fernandofleury de ter um backend API pra isso. Agora, editar o JSON já resolve pra gente :)

O que pensei tbm, é integrar uma ferramenta de formulário (Google Docs, Typeform, etc.) com a ferramenta de CLI. Assim, quem não manja muito de abrir PR e etc. poderia contribuir também. Apesar do público ser dev, tem gente que não manja muito de Github. E mais ainda, teríamos uma forma de validar os dados antes de chegarem na ferramenta.

**Comment by LFeh** [comment]
+1

**Comment by leandrogrillo** [comment]
Eae galera, vamos tocar o projetin, começar a por algo no papel pra termos algo mais palpável?

**Comment by vitorbritto** [comment]
Estou com 3 projetos para liberar. Está corrido por aqui. Posso 
organizar algo no final de semana e repasso aqui. =]

**Comment by leandrogrillo** [comment]
De boas @vitorbritto mandei algo aqui só pra mantermos a ideia viva e não ficar só por aqui!

A ideia era termos sugestões de estruturas e afins, achar a melhor forma. Se a galera quiser sugerir algo, como a estrutura de como vai ser o JSON, posta ae para discutirmos

**Comment by eduardojmatos** [comment]
@leandrogrillo @vitorbritto digo o mesmo que o Vitor. Semaninha foi tensa. Na próxima já consigo ajudar de boas. Seria legal marcar um hangout, como fizemos no de Eventos hein
